### ISSUE TITLE
How to convert a physical point made in one MRI image to the registered modulated normalized space?

### ISSUE THREAD

**Issue by johannesSX** [ChristianGaser/cat12#40]
Hello CAT team,
I have a question. Let's assume you've made annotations in the original MRI images and registered them to an MNI template using CAT12 (following the instructions here: [CATStandalone](https://neuro-jena.github.io/enigma-cat12/#standalone)). CAT returns the wmi*.nii file, among other files.

How can I convert an annotation which was made in the physical space of the original MRI to the physical space of the registered and modulated MRI?

Code example:
```
import torchio as tio
import SimpleITK as sitk
tio_org = tio.ScalarImage('../data/tmp/test/org.nii')
sitk_org = tio_org.as_sitk()

tio_wmi = tio.ScalarImage('../data/tmp/test/wmisorg.nii')
sitk_wmi = tio_wmi.as_sitk()

point_in_org_mri = (sitk_org.GetSize()[0] // 2, sitk_org.GetSize()[1] // 2, sitk_org.GetSize()[2] // 2)
_x, _y, _z = sitk_org.TransformIndexToPhysicalPoint(point_in_org_mri)
point_in_wmi_mri = transform_point(_x, _y, _z) # ????
```

I've imported the MRIs into SimpleITK, but I'd also appreciate a tip for Matlab. How should I proceed? Is there a function to convert points based on one coordinate system to the CAT-normalized coordinate system?

**Comment by johannesSX** [comment]
I now think something is wrong with the displacement field. Here in the Matlab code:

`def = spm_read_vols(spm_vol('y_IXI013-HH-1212-T1.nii')); % CAT deformation field`

I'm not getting a 3-channel image. I'm getting a single-channel image (169, 205, 169).

When I read the displacement field into SimpleITK, I get
`sitk_deformation_field` = sitk.ReadImage('./y_IXI013-HH-1212-T1.nii', sitk.sitkVectorFloat64)
sitk_deformation_field.GetSize()
`
(169, 205, 169, 1, 3). But shouldn't it actually output 3, 1, 169, 205, 169?

In other words: How do I correctly read the displacement field in Matlab?

**Comment by ChristianGaser** [comment]
There seems to be an error in the current CAT12 version that ignores the additional dimensions. We are working on that...

**Comment by johannesSX** [comment]
I solved it. I can't say where my thinking went wrong.
It works now with nibabel... It's not very elegant, though, and I'd be very grateful for suggestions for improvement. But it works now.

```
    import nibabel as nib
    import numpy as np

    # Step 1: Load the deformation field
    deformation_file = '../data/tmp/test/y_IXI013-HH-1212-T1.nii'
    deformation_img = nib.load(deformation_file)
    deformations_data = np.squeeze(deformation_img.get_fdata())

    # Step 2: Load the source MRI image (MRI A)
    mri_a_file = '../data/tmp/test/IXI013-HH-1212-T1.nii.gz'
    mri_a_img = nib.load(mri_a_file)
    mri_a_data = mri_a_img.get_fdata()
    mri_a_affine = mri_a_img.affine

    # Step 3: Define the point in MRI A voxel coordinates
    point_a_voxel = np.array([128, 170, 75])

    # Step 4: Convert the point to mm coordinates in MRI A space
    point_a_mm = nib.affines.apply_affine(mri_a_affine, point_a_voxel)

    # Step 5: Find the corresponding voxel in the deformation field
    def find_closest_voxel(point_mm, deformations_data, tpm_affine):
        tpm_size = deformations_data.shape[:3]
        closest_indices = None
        min_distance = np.inf

        for x in tqdm.tqdm(range(tpm_size[0])):
            for y in range(tpm_size[1]):
                for z in range(tpm_size[2]):
                    mapped_mm = deformations_data[x, y, z, :]

                    distance = np.linalg.norm(point_mm - mapped_mm)
                    if distance < min_distance:
                        min_distance = distance
                        closest_indices = np.array([x, y, z])

        return closest_indices

    tpm_file = '../data/tmp/test/wmiIXI013-HH-1212-T1.nii.gz'
    tpm_img = nib.load(tpm_file)
    tpm_affine = tpm_img.affine

    # Find the closest voxel indices in the TPM space
    closest_voxel_indices = find_closest_voxel(point_a_mm, deformations_data, tpm_affine)

    print("Point in MRI A (voxel coordinates):", point_a_voxel)
    print("Point in MRI A (mm coordinates):", point_a_mm)
    print("Closest voxel indices in TPM:", closest_voxel_indices)
```
